### PR TITLE
feat: set oci labels for docker image in ci job

### LINKED ISSUE #75: OCI annotations for docker image
I propose to convert the LABELs in Dockerfile[^1] to OCI Annotations[^2].

Those would show up nicely rendered here: https://github.com/kube-logging/fluentd-images/pkgs/container/fluentd/128932176?tag=v1.15-staging

Possible implementations[^3]:

- adding labels in `.github/workflows/ci.yml`
- adding labels in `Dockerfile`

[^1]: https://github.com/kube-logging/fluentd-images/blob/main/v1.16/Dockerfile#L5
[^2]: https://github.com/opencontainers/image-spec/blob/main/annotations.md
[^3]: https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#labelling-container-images

### PR DESCRIPTION
**Pull request by 24367dfa**
closes #75

### CODE REVIEW

**Review by aslafy-z — APPROVED** [review]

**Comment by pepov** [comment]
@24367dfa something is off with the commits, can you please rebase your change against the main branch?

**Comment by 24367dfa** [comment]
> @24367dfa something is off with the commits, can you please rebase your change against the main branch?

is off was lightly put... i am sorry for the inconvenience. should be fixed now.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -29,6 +29,12 @@ jobs:
           tags: |
             type=raw,value=${{ matrix.fluentd }}-build.${{ github.run_number }}
             type=raw,value=${{ matrix.fluentd }}
+          labels: |
+            org.opencontainers.image.description=Custom-built Fluentd image for the Logging operator
+            org.opencontainers.image.title=Fluentd docker image
+            org.opencontainers.image.authors=kube-logging authors
+            org.opencontainers.image.documentation=https://kube-logging.dev/docs/
+            org.opencontainers.image.version=${{ matrix.fluentd }}
```
[the remark this answers is not in the document]
Comment (OverOrion) [reply .github/workflows/ci.yaml:37]: I think that it is fine as the gemfile's fluentd version should be in sync with the matrix anyway.

File: v1.16/Dockerfile [hunk 2]
```diff
@@ -1,9 +1,5 @@
 FROM alpine:3.18
 
-ENV FLUENTD_VERSION 1.16.2
```
Comment (OverOrion) [inline v1.16/Dockerfile:3]: Is it really safe to delete the `env` variable?

**Review by OverOrion — COMMENTED** [review]
The OCI labels part looks good to me, not sure about the env var removal though.

File: .github/workflows/ci.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pepov) [reply .github/workflows/ci.yaml:37]: I agree, this is fine

File: v1.16/Dockerfile (hunk 2 above)
Comment (pepov) [reply v1.16/Dockerfile:3]: I'm not sure either, so please keep it in case something builds on it.

Comment (24367dfa) [reply v1.16/Dockerfile:3]: i see it as a possible source for inconsistencies, because this variable won´t get updated by renovate.

**Review by pepov — APPROVED** [review]

**Pull request merged**